src/Voron/Data/CompactTrees/CompactTree.cs

```diff
@@ -721,6 +734,18 @@ public void Add(ReadOnlySpan<byte> key, long value)
             var encodedKey = FindPageFor(key, ref _internalCursor);
             AddToPage(encodedKey, value);
         }
+        
+        public void Add(ReadOnlySpan<byte> key, long value, EncodedKey encodedKey)
+        {
+            if (value < 0)
+                throw new ArgumentOutOfRangeException(nameof(value), value, "Only positive values are allowed");
+            if (key.Length > 1024)
```

Comment:
`1024` - make it const

Maybe move the validation to dedicated method?

Reply:
Done.